trunk/source/CACSL2BoogieTranslator/src/de/uni_freiburg/informatik/ultimate/cdt/translation/implementation/base/library/LibraryModelHandler.java

```diff
@@ -0,0 +1,167 @@
+import org.eclipse.cdt.core.dom.ast.IASTFunctionDefinition;
+import org.eclipse.cdt.core.dom.ast.IASTIdExpression;
+import org.eclipse.cdt.core.dom.ast.IASTNode;
+
+import de.uni_freiburg.informatik.ultimate.cdt.translation.implementation.FlatSymbolTable;
+import de.uni_freiburg.informatik.ultimate.cdt.translation.implementation.LocationFactory;
+import de.uni_freiburg.informatik.ultimate.cdt.translation.implementation.base.IDispatcher;
+import de.uni_freiburg.informatik.ultimate.cdt.translation.implementation.base.TranslationSettings;
+import de.uni_freiburg.informatik.ultimate.cdt.translation.implementation.container.c.ICType;
+import de.uni_freiburg.informatik.ultimate.cdt.translation.implementation.exception.UnsupportedSyntaxException;
+import de.uni_freiburg.informatik.ultimate.cdt.translation.implementation.result.Result;
+import de.uni_freiburg.informatik.ultimate.core.model.models.ILocation;
+import de.uni_freiburg.informatik.ultimate.core.model.services.ILogger;
+
+/**
+ * The {@link LibraryModelHandler} creates the translation for various functions and types where we have our own
+ * specification or implementation. This is typically the case for functions defined in the C standard, but also for
+ * various standard libraries or SV-COMP extensions.
+ *
+ * @author Markus Lindenmann,
+ * @author Matthias Heizmann
+ * @author Daniel Dietsch (dietsch@informatik.uni-freiburg.de)
+ * @author Frank Schüssele (schuessf@informatik.uni-freiburg.de)
+ */
+public class LibraryModelHandler {
+	private final LocationFactory mLocationFactory;
+	private final Map<String, IFunctionModelHandler> mFunctionModels;
+	private final Map<String, ICType> mTypeModels;
+	private final Map<String, IASTNode> mFunctionTable;
+	private final FlatSymbolTable mSymboltable;
+	private final TranslationSettings mSettings;
+	private final ILogger mLogger;
+
+	public LibraryModelHandler(final ILogger logger, final Map<String, IASTNode> functionTable,
+			final FlatSymbolTable symboltable, final TranslationSettings settings,
+			final LocationFactory locationFactory, final List<ILibraryModel> libraryModels) {
+		mLogger = logger;
+		mFunctionTable = functionTable;
+		mSymboltable = symboltable;
+		mSettings = settings;
+		mLocationFactory = locationFactory;
+		mFunctionModels = getFunctionModels(libraryModels);
+		mTypeModels = getTypeModels(libraryModels);
+	}
+
+	/**
+	 * Check if the given function has an "integrated" specification or implementation and return a {@link Result} that
+	 * contains a translation of the function if this is the case.
+	 *
+	 * Return null otherwise.
+	 *
+	 * @param main
+	 * @param node
+	 * @param astIdExpression
+	 * @return
+	 */
+	public Result translateStandardFunction(final IDispatcher main, final IASTFunctionCallExpression node,
+			final IASTIdExpression astIdExpression) {
+		assert node.getFunctionNameExpression() == astIdExpression
+				: "astIdExpression is not the name of the called function";

```

Comment:
If this is required, why pass two parameters at all?

Reply:
Good point, I simply copied this from `StandardFunctionHandler`, but I will check it again.

Reply:
This argument was indeed not necessary, therefore I removed it (740c4ea).